### PR TITLE
fix: add url check and wrapping for commit msg field

### LINKED ISSUE #7158: Message on Synced -> More field popping out
If you are trying to resolve an environment-specific issue or have a one-off question about the edge case that does not require a feature then please consider asking a question in argocd slack [channel](https://argoproj.github.io/community/join-slack).

Checklist:

*[ *] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [ *] I've included steps to reproduce the bug.
* [ *] I've pasted the output of `argocd version`.

**Describe the bug**
1. Message content not get wrapped into More box
2. We are providing link to the code change in commit message. Currently the link in the message not clickable, is it possible to make it clickable?

**To Reproduce**

Commit message with more than 100 or 200 words.

**Expected behavior**

Message should be wrapped in the white box and link inside the commit message should be clickable.

**Screenshots**
image

**Version**

```shell
v2.1.2+7af9dfb
```

**Logs**

```
Paste any relevant application logs here.
```

### PR DESCRIPTION
**Pull request by saumeya**
Signed-off-by: saumeya <saumeyakatyal@gmail.com>

closes #7158 
Screenshot from 2021-09-24 13-04-09

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: ui/src/app/applications/components/application-details/application-details.tsx [hunk 1]
```diff
@@ -146,6 +146,18 @@ export class ApplicationDetails extends React.Component<RouteComponentProps<{nam
                                 return this.filterTreeNode(resNode, treeFilter);
                             });
 
+                            const URL_REGEX = /https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\+.~#?&//=]*)/;
```
Comment (alexmt) [inline ui/src/app/applications/components/application-details/application-details.tsx:149]: Nitpick: We already have URL regex defined in application-summary.tsx:


https://github.com/argoproj/argo-cd/blob/master/ui/src/app/applications/components/application-summary/application-summary.tsx#L19

Can you please move it to https://github.com/argoproj/argo-cd/blob/master/ui/src/app/applications/components/utils.tsx and re-use in both application-summary.tsx and application-details.tsx

**Review by alexmt — CHANGES_REQUESTED** [review]
Thank you @saumeya ! Added one small comment, can you please take a look?

File: ui/src/app/applications/components/application-details/application-details.tsx (hunk 1 above)
Comment (saumeya) [reply ui/src/app/applications/components/application-details/application-details.tsx:149]: sure

**Review by alexmt — APPROVED** [review]
LGTM!

**Pull request merged**